common/utils.ts

```diff
@@ -17,21 +18,88 @@ import {
   DATE_FORMAT_PATTERN,
   COMPONENT_CATEGORY,
   NODE_CATEGORY,
+  WorkspaceFormValues,
 } from './';
 
 // TODO: implement this and remove hardcoded return values
 /**
- * Converts a ReactFlow workspace flow to a backend-compatible set of ingest and/or search sub-workflows,
- * along with a provision sub-workflow if resources are to be created.
+ * Given a ReactFlow workspace flow and the set of current form values within such flow,
+ * generate a backend-compatible set of sub-workflows.
+ *
  */
 export function toTemplateFlows(
-  workspaceFlow: WorkspaceFlowState
+  workspaceFlow: WorkspaceFlowState,
+  formValues: WorkspaceFormValues
 ): TemplateFlows {
+  const textEmbeddingTransformerNodeId = Object.keys(formValues).find((key) =>
+    key.includes('text_embedding')
```

Comment:
Constant for `text_embedding`?

Reply:
Yeah this entire fn will be more generic. Just a first set of hardcoded vals to get e2e working.

Reply:
See TODO on line 24.